Разработаны все тесты по TMatrix

McNikidry opened the pull request.

McNikidry committed:
Разработаны все тесты по TMatrix

alvls commented on test/test_tmatrix.cpp:
```diff
@@ -1,151 +1,223 @@
+TEST(TMatrix, copied_matrix_is_equal_to_source_one)
+{
+	TMatrix<int> m(4);
+	TMatrix<int> m1 = m;
```
Сначала измените матрицу, потом копируйте.

alvls commented on test/test_tmatrix.cpp:
```diff
@@ -1,151 +1,223 @@
+
+TEST(TMatrix, can_assign_matrix_to_itself)
+{
+	TMatrix<int> m(4);
```
Неверный тест. Нужно проверить присваивание m = m

alvls commented on test/test_tmatrix.cpp:
```diff
@@ -1,151 +1,223 @@
+	TMatrix<int> m(size);
+	TMatrix<int> m1(size);
+
+	EXPECT_EQ(m, m1);
```
Неверный тест. Где присваивание?

alvls commented on test/test_tmatrix.cpp:
```diff
@@ -1,151 +1,223 @@
+	int size = 4;
+
+	TMatrix<int> m(size);
+	TMatrix<int> m1(size + 1);
```
Неверный тест. Где присваивание?

alvls commented on test/test_tmatrix.cpp:
```diff
@@ -1,151 +1,223 @@
+	TMatrix<int> m(size);
+	TMatrix<int> m1(size);
+
+	EXPECT_EQ(m, m1);
```
Лучше явно вызвать оператор ==

McNikidry replied:
А для того, чтобы проверить на ==, какой макрос применять. Знаю, что есть какой-то на true

alvls replied:
EXPECT_EQ(true, m==m1)

alvls commented on test/test_tmatrix.cpp:
```diff
@@ -1,151 +1,223 @@
+
+	TMatrix<int> m(size);
+
+	EXPECT_EQ(m, m);
```
Лучше явно вызвать оператор ==

McNikidry committed:
Исправил замечания по тестам

Если сейчас будет все верно, то мне показывать все равно лабораторную на следующей паре?
